exporters/otlp/src/otlp_grpc_exporter.cc

```diff
@@ -23,22 +24,53 @@ namespace otlp
 
 OtlpGrpcExporter::OtlpGrpcExporter() : OtlpGrpcExporter(OtlpGrpcExporterOptions()) {}
 
-OtlpGrpcExporter::OtlpGrpcExporter(const OtlpGrpcExporterOptions &options)
-    : options_(options),
-#ifdef ENABLE_ASYNC_EXPORT
-      client_(std::make_shared<OtlpGrpcClient>()),
-#endif
-      trace_service_stub_(OtlpGrpcClient::MakeTraceServiceStub(options))
-{}
+OtlpGrpcExporter::OtlpGrpcExporter(const OtlpGrpcExporterOptions &options) : options_(options)
+{
+  client_                 = OtlpGrpcClientFactory::Create(options_);
+  client_reference_guard_ = OtlpGrpcClientFactory::CreateReferenceGuard();
+  client_->AddReference(*client_reference_guard_, options_);
```

Comment:
Create the reference guard with 1 ref count? So we don't need add reference right after creation.

Reply:
The lifetime of `client_reference_guard_` is tied to the exporters. It might be better to handle reference counts only through the `OtlpGrpcClient`‘s APIs. Alternatively, we could create a common base class for all gRPC exporters to prevent missing any `AddReference` calls. What do you think?